.github/workflows/check-learning-path-links.yml

```diff
@@ -0,0 +1,79 @@
+name: 'Check Learning Path Links'
+on:
+  schedule: # Run once a month
+    - cron: '0 0 1 * *'
+  workflow_dispatch:
+
+permissions: {}
+  
+jobs:
+  check-learning-path-links:
+    name: 'Check Learning Path Links'
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+
+    steps:
+      - name: Checkout head
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+          fetch-depth: 0 # Fetch the entire repo for the below git commit graph operations
+          ref: main
+
+      - name: Get current SHA
+        id: get_new_sha
+        run: |
+          new_sha=$(git rev-parse HEAD)
+          echo "new_sha=$new_sha" >> $GITHUB_ENV
+
+      - name: Get previous update SHA
+        id: get_sha
+        run: |
+          cd .github
+          prev_sha=$(cat learning-path-sha.txt)
+          echo "prev_sha=$prev_sha" >> $GITHUB_ENV
+          cd ../
```

Comment:
Not necessary to change the working directory here, we can just use:
`prev_sha=$(cat .github/learning-path-sha.txt)`